app/javascript/App.jsx

```diff
@@ -12,8 +12,11 @@ export default function App() {
   const containerHeight = currentUser?.signed_in ? 'full-height' : 'h-100';
 
   // Greenlight V3 brand-color theming
-  const { isLoading, data: primaryColor } = useSiteSettingAsync('PrimaryColor');
-  document.documentElement.style.setProperty('--brand-color', primaryColor);
+  const { isLoading, data: brandColor } = useSiteSettingAsync('PrimaryColor');
```

Comment:
From a JS perspective `useSiteSetting` and `useSiteSettingAsync` are exactly the same.
The only difference that made it work is that you're returning the resolved value data property `promise.data` when the other is returning the resolved value data data property 🤣 `.then((resp) => resp.data.data),`.

When the actual `SiteSettings#show` **API** isn't encapsulating the setting value inside of a data root property so it's returning `<value>` instead of `{"data": <value>}`